gnomad_qc/v3/resources/meta.py

```diff
@@ -1,37 +1,113 @@
+                                              CURRENT_PROJECT_META_VERSION,
+                                              CURRENT_RELEASE, RELEASES)
+
 
 # Samples metadata
-META_ROOT = "gs://gnomad/metadata/genomes_v3"
-meta = TableResource(f'{META_ROOT}/gnomad_v3_metadata_2019-09-27.ht')
-meta_tsv_path = f'{META_ROOT}/gnomad_v3_metadata_2019-09-27.tsv.gz'
-project_meta = TableResource(
-    import_func=hl.import_table,
-    import_args={
-        'path': f'{META_ROOT}/09-09-2019_v3_project_meta.txt',
-        'impute': True,
-        'key': 's',
-        'min_partitions': 100
-    }
-)
-pedigree = VersionedPedigreeResource(
-    'final',  # TODO: Make sure "final" is the best label once the family scripts are in
-    {
-        'raw': PedigreeResource(f'{META_ROOT}/gnomad_v3_raw.fam', delimiter="\t"),
-        'final': PedigreeResource(f'{META_ROOT}/gnomad_v3.fam', delimiter="\t")
-    }
-)
+def _meta_root_path(version: str = CURRENT_RELEASE) -> str:
+    """
+    Retrieves the path to the root metadata directory
+
+    :param version: gnomAD release version
+    :return: String representation of the path to the root metadata directory
+    """
+    return f"gs://gnomad/metadata/genomes_v{version}"
+
+
+def meta_tsv_path(
+    version: str = CURRENT_RELEASE, meta_version: str = CURRENT_META_VERSION
+) -> str:
+    """
+    Gets the path to the finalized sample metadata information after sample QC
+
+    :param version: gnomAD release version
+    :param meta_version: metadata version to return
+    :return: String path to the finalized metadata
+    """
+    return (
+        f"{_meta_root_path(version)}/gnomad_v{version}_metadata_{meta_version}.tsv.gz"
+    )
+
 
-trios = VersionedPedigreeResource(  # TODO: Should this be merged with Pedigree into a single resource?
-    'final',  # TODO: Make sure "final" is the best label once the family scripts are in
+_meta_versions = {
+    "2019-09-27": TableResource(
+        path="gs://gnomad/metadata/genomes_v3/gnomad_v3_metadata_2019-09-27.ht"
+    ),
+    "2019-09-25": TableResource(
+        path="gs://gnomad/metadata/genomes_v3/gnomad_v3_metadata_2019-09-25.ht"
+    ),
+}
+_meta_versions["v3"] = _meta_versions["2019-09-27"]
+
+
+_project_meta_versions = {
+    "2020-09-11": TableResource(
+        path="gs://gnomad/metadata/genomes_v3.1/2020-09-11_v3.1_project_meta.ht"
+    ),
+    "09-09-2019": TableResource(
+        path="gs://gnomad/metadata/genomes_v3/09-09-2019_v3_project_meta.ht",
+        import_func=hl.import_table,
+        import_args={
+            "paths": "gs://gnomad/metadata/genomes_v3/09-09-2019_v3_project_meta.txt",
+            "impute": True,
+            "key": "s",
+            "min_partitions": 100,
+        },
+    ),
+}
+_project_meta_versions["v3"] = _project_meta_versions["09-09-2019"]
+_project_meta_versions["v3.1"] = _project_meta_versions["2020-09-11"]
+
+
+_pedigree_versions = {
+    "final": PedigreeResource(
```

Comment:
Why use "final" here vs "v3.1"? Same question for `_trios_versions`.

Reply:
I was trying to reduce changes to the code, e.g. L103 and L104, moving forward but I think the trade off with the current approach is less clarity and consistency so I'll update to "v3.1" for both.